File: backend/app/api/v1/brands.py
from fastapi import APIRouter, Depends, Query
from sqlalchemy.ext.asyncio import AsyncSession

from app.core.database import get_db
from app.core.dependencies import get_current_user, require_roles
from app.schemas.brand import BrandCreate, BrandRead, BrandUpdate
from app.services import brand_service

router = APIRouter(prefix="/brands", tags=["brands"])


@router.get("", response_model=list[BrandRead])
async def list_brands(
    active_only: bool = Query(False),
    db: AsyncSession = Depends(get_db),
    _user=Depends(get_current_user),
):
    return await brand_service.list_brands(db, active_only=active_only)


@router.post("", response_model=BrandRead, status_code=201)
async def create_brand(
    body: BrandCreate,
    db: AsyncSession = Depends(get_db),
    _admin=Depends(require_roles("admin")),
):
    return await brand_service.create_brand(db, body)


@router.get("/{brand_id}", response_model=BrandRead)
async def get_brand(
    brand_id: str,
    db: AsyncSession = Depends(get_db),
    _user=Depends(get_current_user),
):
    return await brand_service.get_brand(db, brand_id)


@router.patch("/{brand_id}", response_model=BrandRead)
async def update_brand(
    brand_id: str,
    body: BrandUpdate,
    db: AsyncSession = Depends(get_db),
    _admin=Depends(require_roles("admin")),
):
    return await brand_service.update_brand(db, brand_id, body)


@router.delete("/{brand_id}", status_code=204)
async def delete_brand(
    brand_id: str,
    db: AsyncSession = Depends(get_db),
    _admin=Depends(require_roles("admin")),
):
    await brand_service.delete_brand(db, brand_id)

File: frontend/src/api/brands.ts
import client from './client';
import type { Brand, BrandCreate, BrandUpdate } from '../types/brand';

export async function listBrands(activeOnly = false): Promise<Brand[]> {
  const response = await client.get<Brand[]>('/brands', { params: { active_only: activeOnly } });
  return response.data;
}

export async function getBrand(id: string): Promise<Brand> {
  const response = await client.get<Brand>(`/brands/${id}`);
  return response.data;
}

export async function createBrand(data: BrandCreate): Promise<Brand> {
  const response = await client.post<Brand>('/brands', data);
  return response.data;
}

export async function updateBrand(id: string, data: BrandUpdate): Promise<Brand> {
  const response = await client.patch<Brand>(`/brands/${id}`, data);
  return response.data;
}

export async function deleteBrand(id: string): Promise<void> {
  await client.delete(`/brands/${id}`);
}
